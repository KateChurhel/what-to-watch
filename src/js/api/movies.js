/* istanbul ignore file */
import { sendGetRequest } from './index';

export const getCategoryData = (category, data) => sendGetRequest(`/${category}/popular`, data);

export const getDetailsData = (category, id, data) => sendGetRequest(`/${category}/${id}`, data);

export const getSearchResult = (category, data) => sendGetRequest(`/search/${category}`, data);

export const getDiscoverResult = (data) => sendGetRequest('/discover/movie', data);

export const getGenres = () => sendGetRequest('/genre/movie/list');
