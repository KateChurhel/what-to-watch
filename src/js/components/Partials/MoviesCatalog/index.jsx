// libraries
import React from 'react';
import { useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
// helpers
import { formatCardData } from '../../../helpers/formatData';
// views
import SmallMovieCard from '../SmallMovieCard';

const MoviesCatalog = ({ catalogData = [] }) => {
  const { category } = useParams();

  return (
    <div className="catalog__movies-list">
      {
        catalogData.map(
          (cardProps) => (
            <SmallMovieCard
              {...formatCardData(cardProps)}
              key={cardProps.id}
              link={`/details/${category}/${cardProps.id}`}
            />
          ),
        )
      }
    </div>
  );
};

MoviesCatalog.propTypes = {
  catalogData: PropTypes.arrayOf(PropTypes.shape({})),
};

export default MoviesCatalog;
