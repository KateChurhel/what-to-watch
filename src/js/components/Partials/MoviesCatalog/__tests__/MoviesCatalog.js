import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import MoviesCatalog from '../index';
import CATEGORIES from '../../../../constants/categories';
import ROUTES from '../../../../constants/routes';
import { MOCK_MOVIES } from '../../../../mocks/movies';

describe('Movies Catalog: ', () => {
  it('correctly renders with empty result', async () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={[`/${CATEGORIES.movie}/1`]} initialIndex={1}>
        <Route path={ROUTES.catalog}>
          <MoviesCatalog />
        </Route>
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('correctly renders with 1 result', async () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={[`/${CATEGORIES.movie}/1`]} initialIndex={1}>
        <Route path={ROUTES.catalog}>
          <MoviesCatalog catalogData={MOCK_MOVIES.slice(0, 3)} />
        </Route>
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
