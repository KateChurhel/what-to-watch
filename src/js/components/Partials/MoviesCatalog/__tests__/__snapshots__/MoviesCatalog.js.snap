// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Catalog:  correctly renders with 1 result 1`] = `
<DocumentFragment>
  <div
    class="catalog__movies-list"
  >
    <a
      class="small-movie-card catalog__movies-card small-movie-card__link"
      href="/details/movie/313369"
    >
      <div
        class="small-movie-card__image "
      >
        <img
          alt="La La Land"
          height="450"
          loading="lazy"
          src="https://image.tmdb.org/t/p/w300/uDO8zWDhfWwoFdKS4fzkUJt0Rf0.jpg"
          width="300"
        />
      </div>
      <div
        class="small-movie-card__info"
      >
        <h3
          class="small-movie-card__title"
        >
          La La Land
        </h3>
      </div>
    </a>
    <a
      class="small-movie-card catalog__movies-card small-movie-card__link"
      href="/details/movie/516486"
    >
      <div
        class="small-movie-card__image "
      >
        <img
          alt="Greyhound"
          height="450"
          loading="lazy"
          src="https://image.tmdb.org/t/p/w300/kjMbDciooTbJPofVXgAoFjfX8Of.jpg"
          width="300"
        />
      </div>
      <div
        class="small-movie-card__info"
      >
        <h3
          class="small-movie-card__title"
        >
          Greyhound
        </h3>
      </div>
    </a>
    <a
      class="small-movie-card catalog__movies-card small-movie-card__link"
      href="/details/movie/385103"
    >
      <div
        class="small-movie-card__image "
      >
        <img
          alt="Scoob!"
          height="450"
          loading="lazy"
          src="https://image.tmdb.org/t/p/w300/jHo2M1OiH9Re33jYtUQdfzPeUkx.jpg"
          width="300"
        />
      </div>
      <div
        class="small-movie-card__info"
      >
        <h3
          class="small-movie-card__title"
        >
          Scoob!
        </h3>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`Movies Catalog:  correctly renders with empty result 1`] = `
<DocumentFragment>
  <div
    class="catalog__movies-list"
  />
</DocumentFragment>
`;
